File: decompiled/sources/com/AppManager.java
package com;

import android.app.Activity;
import java.util.Stack;

public class AppManager {
    private static Stack<Activity> activityStack;
    private static AppManager instance;

    public static AppManager getAppManager() {
        if (instance == null) {
            instance = new AppManager();
        }
        return instance;
    }

    public void addActivity(Activity activity) {
        if (activityStack == null) {
            activityStack = new Stack<>();
        }
        activityStack.add(activity);
    }

    public void finishAllActivity() {
        for (int i = 0; i < activityStack.size(); i++) {
            if (activityStack.get(i) != null) {
                ((Activity) activityStack.get(i)).finish();
            }
        }
        activityStack.clear();
    }
}

File: decompiled/sources/com/AppManager.kt
package com

import android.app.Activity
import java.util.Stack

class AppManager private constructor() {

    fun addActivity(activity: Activity) {
        if (activityStack == null) {
            activityStack = Stack()
        }
        activityStack!!.add(activity)
    }

    fun finishAllActivity() {
        val stack = activityStack ?: return
        for (index in stack.indices) {
            val activity = stack[index]
            activity?.finish()
        }
        stack.clear()
    }

    companion object {
        private var activityStack: Stack<Activity?>? = null
        private var instance: AppManager? = null

        @JvmStatic
        fun getAppManager(): AppManager {
            if (instance == null) {
                instance = AppManager()
            }
            return instance!!
        }
    }
}
